tests/test-solve.C

```diff
@@ -48,78 +48,58 @@ using namespace FFPACK;
 using Givaro::Modular;
 using Givaro::ModularBalanced;
 
-template<typename T>
-void write_matrix(Givaro::Integer p, size_t m, size_t n, T* C, size_t ldc){
-
-	size_t www=(p.bitsize()*log(2.))/log(10.);
-	for (size_t i=0;i<m;++i){
-		cout<<"[ ";
-		cout.width(www+1);
-		cout<<std::right<<C[i*ldc];
-		for (size_t j=1;j<n;++j){
-			cout<<" ";
-			cout.width(www);
-			cout<<std::right<<C[i*ldc+j];
-		}
-		cout<<"]"<<endl;
-	}
-	cout<<endl;
-
-}
-
 
 template<typename Field, class RandIter>
 bool check_solve(const Field &F, size_t m, RandIter& Rand){
 
-	typedef typename Field::Element Element;
-	Element * A, *A2,  *B, *B2, *x;
+typedef typename Field::Element_ptr Element;
```

Comment:
Don't do that: it is misleading as the typedef'ed Element is not an alias to Field::Element but Field::Element_ptr.